### PR TITLE
[cluster test] Allow to restart instance with different tag

### PR DESCRIPTION
**Pull request by andll**
* kube swarm now maps node by pod_name, not by InstanceConfig. This allows to change instance config and restart instance on same node
* Introducing Instance::instance_config to grab instance config of this instance
* Introducing InstanceConfig::replace_tag to update tag on InstanceConfig

### CODE REVIEW

File: testsuite/cluster-test/src/instance.rs [hunk 1]
```diff
@@ -80,6 +80,42 @@ struct K8sInstanceInfo {
     kube: ClusterSwarmKube,
 }
 
+impl InstanceConfig {
+    pub fn replace_tag(&mut self, new_tag: String) -> Result<()> {
+        match self {
```
Comment (ankushagarwal) [inline testsuite/cluster-test/src/instance.rs:85]: since self is already &mut, I don't think you need `ref mut` in the match handles

Comment (andll) [reply testsuite/cluster-test/src/instance.rs:85]: Yep, you are right

**Pull request closed without merging**